core/src/stdlib/pyscript/events.py

```diff
@@ -0,0 +1,165 @@
+import asyncio
+import inspect
+import sys
+
+from functools import wraps
+from pyscript.magic_js import document
+from pyscript.ffi import create_proxy
+from pyscript.util import is_awaitable
+
+
+class Event:
+    """
+    Represents something that may happen at some point in the future.
+    """
+
+    def __init__(self):
+        self._listeners = []
+
+    def trigger(self, result):
+        """
+        Trigger the event with a result to pass into the handlers.
+        """
+        for listener in self._listeners:
+            if is_awaitable(listener):
+                # Use create task to avoid making this an async function.
+                asyncio.create_task(listener(result))
+            else:
+                listener(result)
+
+    def add_listener(self, listener):
+        """
+        Add a callable/awaitable to listen to when this event is triggered.
+        """
+        if is_awaitable(listener) or callable(listener):
+            if listener not in self._listeners:
+                self._listeners.append(listener)
+        else:
+            raise ValueError("Listener must be callable or awaitable.")
```

Comment:
technically speaking, a listener can be an instance with a `handleEvent(event)` method in it too ... it's a quite popular approach for libraries' authors in the JS world, not sure we should limit this implementation, although it's admittedly less common or less known as a pattern but it's full of potentials in general.

Reply:
```python
class Listener:
    def handleEvent(self, event):
        print(event.type)

from pyscript import document

document.body.addEventListener("click", Listener())
```

This *should* work but I've just realized it currently doesn't ... never mind then, let's keep it simple for the time being.

Reply:
Yeah, if folks ask for it from a Python PoV, we can fix it.